### PR TITLE
docs: add group policies for user auth (replaces 109)

### PR DESCRIPTION
**Pull request by mbshields**
**What type of PR is this?**

documentation

**What does this PR do / Why do we need it**:

Adds ability to define groups for authz policies
This PR supersedes previous PR https://github.com/project-zot/project-zot.github.io/pull/109

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by mbshields** [comment]
Replacing the previous PR 109 because of git issues beyond my ability to easily fix.

To preserve Andrei's review comments on that PR, they are repeated here and (partially?) addressed in the commit.

"These ideas:
- Just like the users, groups can be part of repository policies/default policies/admin policies. This may already be captured in current text.
- When a user logs in the the permissions priority when deciding if he should have permissions on a repo is user>group>default>admin policy. If the user permissions for that repo are set, those are applied, if there are not user permissions, any group permissions set are applied, if there are no user or group permissions set for him, then defaults are applied. If defaults are missing too, then the admin policy is set, which verifies if the user is admin and applies those permissions.
- It works with LDAP too, and the group attribute name is configurable. The DN of the group is used as the group name and the functionality is the same. -> Maybe we should add an example for LDAP. See this config, with these specific lines: https://github.com/project-zot/zot/blob/main/examples/config-ldap.json#L21 and the group "cn=ldap-group,ou=Groups,dc=example,dc=org" mentioned at https://github.com/project-zot/zot/blob/main/examples/config-ldap.json#L34
- If we have an LDAP configuration example somewhere in the documentation maybe we should add missing settings from https://github.com/project-zot/zot/blob/main/examples/config-ldap.json#L15 to the example configuration there.
"

File: docs/articles/authn-authz.md [hunk 1]
```diff
@@ -138,6 +139,7 @@ authentication.
 | `startTLS`      | Set to `true` to enable TLS communication with the LDAP server.                  |
 | `baseDN`        | Starting location within the LDAP directory for performing user searches.        |
 | `userAttribute` | Attribute name used for a user.                                                  |
```
Comment (andaaron) [inline docs/articles/authn-authz.md:141]:
```suggestion
| `userAttribute` | Attribute name used to obtain the username.                                                  |
```

File: docs/articles/authn-authz.md [hunk 2]
```diff
@@ -138,6 +139,7 @@ authentication.
...
+| `userGroupAttribute` | Attribute name used for a group.                                            |
```
Comment (andaaron) [inline docs/articles/authn-authz.md:142]:
```suggestion
| `userGroupAttribute` | Attribute name used to obtain groups a user belongs to.                                            |
```

Comment (mbshields) [reply docs/articles/authn-authz.md]: Changes made

**Review by andaaron — APPROVED** [review]

**Pull request merged**